### DISCUSSION TITLE
quickly adding character or label elements at specified coordinates?

### DISCUSSION THREAD

**Discussion by cqwufly** [jbengler/tidyplots#38] [Propose a feature]
Could you add a function like add_text that allows for quickly adding character or label elements at specified coordinates?

**Comment by jbengler** [comment]
Hi @cqwufly 

Thank you for using tidyplots!

Have you tried the `add_data_labels()` function? Also see https://jbengler.github.io/tidyplots/reference/add_data_labels.html

Best
Jan

**Comment by jbengler** [comment]
@cqwufly Did you have the chance to have a look at my suggestion?

**Reply by cqwufly** [reply]
Thank you for your advice. The add_data_labels()  function can only add labels for each known data point, but sometimes it is necessary to add a label in a specific location. ( For example, add to the boxplot: x=1,y=6,label="test")。through the ggplot2 package geom_text function can be done this function, perhaps tidyplots can have a more simple and elegant way.

**Comment by jbengler** [comment]
I see, you are looking for annotations that do not come from the `data` itself.

Here are some toy examples

```
library(tidyverse)
library(tidyplots)

study %>% 
  tidyplot(x = treatment, y = score, color = treatment) %>% 
  add_boxplot()

# does not really work
study %>% 
  tidyplot(x = treatment, y = score, color = treatment) %>% 
  add_boxplot() %>% 
  add(geom_text(x=1, y=10, label="Some text"))

# quite hacky
study %>% 
  tidyplot(x = treatment, y = score, color = treatment) %>% 
  add_boxplot() %>% 
  add_data_labels(data = data.frame(treatment = 2, score = 22), label="Some text", color = "black")

# use ggplot2 annotate
study %>% 
  tidyplot(x = treatment, y = score, color = treatment) %>% 
  add_boxplot() %>% 
  add(annotate("text", x = 2, y = 20, label = "Some text"))

# make the font 7 point
study %>% 
  tidyplot(x = treatment, y = score, color = treatment) %>% 
  add_boxplot() %>% 
  add(annotate("text", x = 2, y = 20, label = "Some text", size = 7/ggplot2::.pt))
```

**Comment by jbengler** [comment]
Probably, I could provide a wrapper around `ggplot2::annotate()`
https://ggplot2.tidyverse.org/reference/annotate.html

Something like
`add_annotation_text()`
`add_annotation_rectangle()`
`add_annotation_segment()`

Do you think this would make sense?

**Reply by cqwufly** [reply]
study %>% 
  tidyplot(x = treatment, y = score, color = treatment) %>% 
  add_boxplot() %>% 
  add(annotate("text", x = 2, y = 20, label = "Some text"))
Thanks The annotate function feels easier to understand. It's easy to go overboard from the ggplot2 package to tidyplots。providing a wrapper around ggplot2::annotate() should be a good idea!

**Comment by jbengler** [comment]
I just added the following function to the development version of tidyplots

`add_annotation_text()`
`add_annotation_rectangle()`
`add_annotation_segment()`

You can test them by installing the development version like so

```
install.packages("devtools")
devtools::install_github("jbengler/tidyplots")
```

**Reply by cqwufly** [reply]
Thank you for your contribution. These functions are working properly at the moment.
